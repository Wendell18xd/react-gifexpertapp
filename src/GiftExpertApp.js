import React, { useState } from 'react'
import AddCaptegori from './components/AddCaptegori'
import GifGrid from './components/GifGrid'

const GiftExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCaptegori setCategories={setCategories} />
            <hr />

            {
                categories.map(c =>
                    <GifGrid
                        key={c}
                        categories={c}
                    />
                )
            }

        </>
    )
}

export default GiftExpertApp
