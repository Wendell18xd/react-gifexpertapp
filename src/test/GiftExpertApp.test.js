import { shallow } from "enzyme";
import GiftExpertApp from "../GiftExpertApp";

describe('Provando GiftExpertApp', () => {

    test('Renderizando GiftExpertApp', () => {

        const wrapper = shallow(<GiftExpertApp />)
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragonball']
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })


})
