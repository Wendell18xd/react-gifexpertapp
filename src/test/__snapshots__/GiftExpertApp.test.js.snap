// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provando GiftExpertApp Renderizando GiftExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCaptegori
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Provando GiftExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCaptegori
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    categories="One Punch"
    key="One Punch"
  />
  <GifGrid
    categories="Dragonball"
    key="Dragonball"
  />
</Fragment>
`;
